src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -208,6 +301,38 @@ public boolean checkfirepoint(int row, int col){
         return false;
     }
 
+    public boolean checkAIhit(int row, int col){
+        int hitsize = playerHits.size();
+        int i = 0;
+        while( i < hitsize) {
+            Coordinate z = playerHits.get(i);
+            int xhit = z.Across;
+            int yhit = z.Down;
+            if (row == xhit && col == yhit){
+                return true;
+            }else{
+                i += 1;
+            }
+        }
+        return false;
+    }
+
+    public boolean checkAImiss(int row, int col){
+        int missize = playerMisses.size();
```

Comment:
missSize might be a better name for this to keep it like our other variable names